src/js/common/timezone.js

```diff
@@ -136,6 +170,89 @@ function useLocalTimeConverter(arg) {
     return arg instanceof Date || (typeof arg) === 'string';
 }
 
+/**
+ * Extract date tokens (y, M, d, h, m, s) using the formatToParts() method.
+ * @param {Intl.DateTimeFormat} dtf - Intl.DateTimeFormat instance
+ * @param {Date} date - date object
+ * @returns {Array.<number>} An array of objects only containing the formatted date
+ */
+function partsOffset(dtf, date) {
+    var formatted = dtf.formatToParts(date);
+    var filled = [];
+    var i, pos;
+
+    for (i = 0; i < formatted.length; i += 1) {
+        pos = typeToPos[formatted[i].type];
+
+        if (typeof pos !== 'undefined') {
+            filled[pos] = parseInt(formatted[i].value, 10);
+        }
+    }
+
+    return filled;
+}
+
+/**
+ * The time zone offset is calculated from the difference between the current time and the time in a specific time zone.
+ * @param {Array.<number>} parts - An array of objects only containing the formatted date
+ * @param {Date} date - date object
+ * @returns {number} offset
+ */
+function getOffset(parts, date) {
+    var y = parts[0];
+    var M = parts[1];
+    var d = parts[2];
+    var h = parts[3];
+    var m = parts[4];
+    var s = parts[5];
+
+    var utc = new Date(Date.UTC(y, M - 1, d, h, m, s));
+    var offset = (utc - date) / 60 / 1000;
+
+    return Math.round(offset);
+}
+
+/**
+ * Check if browser supports Intl, Intl.DateTimeFormat, formatToParts API
+ * @param {string} timeZone - timezone
+ * @returns {boolean} supported
+ */
+function supportIntl(timeZone) {
+    var supported = false;
+
+    if (window.Intl && window.Intl.DateTimeFormat) {
+        intlFormatter[timeZone] = getIntlFormatter(timeZone);
+
+        if (util.isFunction(intlFormatter[timeZone].formatToParts)) {
```

Comment:
This statement should be run with `if (window.Intl && window.Intl.DateTimeFormat)`. Or `getCustomTimezoneOffset` function will be failed at line 73 `if (intlFormatter[primaryTimezoneName])` in case of no `formatToParts`.